### PR TITLE
fix: refactor test for clarinet 2.0

### PR DESCRIPTION
**Pull request by beguene**
update zip script to ignore node_modules/
update metada.json

closes https://github.com/hirosystems/platform/issues/787

### CODE REVIEW

File: examples/clarity-bitcoin/tests/clarity-bitcoin.test.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Cl, cvToJSON, deserializeCV } from "@stacks/transactions";
```
Comment (hugocaillard) [inline examples/clarity-bitcoin/tests/clarity-bitcoin.test.ts:1]:
```suggestion
import { Cl, ClarityType } from "@stacks/transactions";
```

File: examples/counter/tests/counter.test.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+describe("test individual counters", () => {
+  it("retrieves the default count for a new user", () => {
+    const countResponse = simnet.callReadOnlyFn(
+      "counter",
+      "get-count",
+      [Cl.standardPrincipal(address2)],
+      address1
+    );
+    console.log("countResponse", countResponse);
+    expect(countResponse.result).toStrictEqual(Cl.uint(0));
```
Comment (hugocaillard) [inline examples/counter/tests/counter.test.ts:17]:
```suggestion
    expect(countResponse.result).toBeUint(0);
```

File: examples/clarity-bitcoin/tests/clarity-bitcoin.test.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    const result = await simnet.callReadOnlyFn(
+      contractName,
+      "parse-tx",
+      [Cl.buffer(Buffer.from(btcTx.slice(2), "hex"))],
+      deployer
+    );
+
+    const jsonResult = cvToJSON(result.result);
+    expect(jsonResult.success).toBe(true);
```
Comment (hugocaillard) [inline examples/clarity-bitcoin/tests/clarity-bitcoin.test.ts:21]:
```suggestion
    const { result } =  simnet.callReadOnlyFn(
      contractName,
      "parse-tx",
      [Cl.buffer(Buffer.from(btcTx.slice(2), "hex"))],
      deployer
    );

    expect(result).toHaveClarityType(ClarityType.ResponseOk)
```

Except if you did `cvToJSON` for particular reason, I would favor showcasing clarity custom matchers as much as possible

File: examples/counter/tests/counter.test.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    expect(incrementResponse.result).toBeOk(Cl.bool(true));
+
+    // Retrieve the updated count for address2
+    const updatedCountResponse = simnet.callReadOnlyFn(
+      "counter",
+      "get-count",
+      [Cl.standardPrincipal(address2)],
+      address1
+    );
+    expect(updatedCountResponse.result).toStrictEqual(Cl.uint(2));
```
Comment (hugocaillard) [inline examples/counter/tests/counter.test.ts:64]:
```suggestion
    expect(updatedCountResponse.result).toBeUint(2);
```

File: examples/counter/tests/counter.test.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    expect(incrementResponse.result).toBeOk(Cl.bool(true));
+
+    // Retrieve the updated count for address1
+    const updatedCountResponse = simnet.callReadOnlyFn(
+      "counter",
+      "get-count",
+      [Cl.standardPrincipal(address1)],
+      address1
+    );
+    expect(updatedCountResponse.result).toStrictEqual(Cl.uint(1));
```
Comment (hugocaillard) [inline examples/counter/tests/counter.test.ts:37]:
```suggestion
    expect(updatedCountResponse.result).toBeUint(1);
```

File: examples/lightning-swaps/tests/stxswap_v10.test.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+const contractName = "stxswap_v10";
+const preimageHash =
+  "0x4bf5122f344554c53bde2ebb8cd2b7e3d1600ad631c385a5d7cce23c7785459a";
+const preimage = "0x01";
+const amount = 1000;
+const timelock = 5;
+
+describe("stxswap_v10 contract tests", () => {
+  it("ensures that user can lock and claim stx", async () => {
```
Comment (hugocaillard) [inline examples/lightning-swaps/tests/stxswap_v10.test.ts:17]:
```suggestion
  it("ensures that user can lock and claim stx" () => {
```

Comment (hugocaillard) [reply examples/lightning-swaps/tests/stxswap_v10.test.ts:17]: Most of the simnet method are sync. (`mineBlock`, `callPublicFunction`)

File: examples/lightning-swaps/tests/stxswap_v10.test.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        "lockStx",
+        [
+          preimageHash,
+          Cl.uint(amount),
+          Cl.uint(timelock),
+          Cl.standardPrincipal(wallet2),
+        ],
+        wallet1
+      ),
+    ]);
```
Comment (hugocaillard) [inline examples/lightning-swaps/tests/stxswap_v10.test.ts:31]: This fine, but we can also prefer `simnet.callPublicFunction` when there's only 1 transaction in the block.

The await can be removed

File: examples/non-fungible-token/tests/non-fungible-token.test.ts [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+describe("test non-fungible token contract", () => {
+  it("mints a new NFT and retrieves the last token ID", () => {
+    const block = simnet.mineBlock([
+      tx.callPublicFn(
+        "non-fungible-token",
+        "mint",
+        [Cl.standardPrincipal(wallet1)],
+        deployer
+      ),
+    ]);
```
Comment (hugocaillard) [inline examples/non-fungible-token/tests/non-fungible-token.test.ts:18]: Could be `simnet.callPublicFunction`

**Review by hugocaillard — CHANGES_REQUESTED** [review]
A few remarks.

I didn't comment every line but in general:
- I think the console.log can be removed everywhere
- Privilege custom clarity matchers as much as possible
- No need to await `simnet.callPublicFunction` or `simnet.mineBlock`
- `.callPublicFunction` can often be used instead of `mineBlock` (if there's only when transaction in the block)


Each example should have a tsonfig.json as well:
```
{
  "compilerOptions": {
    "target": "ESNext",
    "useDefineForClassFields": true,
    "module": "ESNext",
    "lib": ["ESNext"],
    "skipLibCheck": true,

    "moduleResolution": "bundler",
    "allowImportingTsExtensions": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,

    "strict": true,
    "noImplicitAny": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noFallthroughCasesInSwitch": true
  },
  "include": ["node_modules/@hirosystems/clarinet-sdk/vitest-helpers/src", "tests"]
}
```


Thanks @beguene!

**Comment by beguene** [comment]
Thanks a lot @hugocaillard for the review ! 
Regarding the migration, I prefer the new syntax obviously, it's much cleaner, and easier. Love the `simnet` to simulate everything, it's intuitive. Regarding migrating from the old way, it is sometimes easier to start from scratch actually.
Also `Cl.TypeX` is better than TypeXCV.

(the console.log was for debugging, I will remove them)

File: examples/counter/tests/counter.test.ts [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+describe("test individual counters", () => {
+  it("retrieves the default count for a new user", () => {
+    const countResponse = simnet.callReadOnlyFn(
+      "counter",
+      "get-count",
+      [Cl.standardPrincipal(address2)],
+      address1
+    );
+    console.log("countResponse", countResponse);
```
Comment (hugocaillard) [inline examples/counter/tests/counter.test.ts:16]:
```suggestion

```

File: examples/counter/tests/counter.test.ts [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+  });
+
+  it("increments the count for a user", () => {
+    const incrementResponse = simnet.callPublicFn(
+      "counter",
+      "count-up",
+      [],
+      address1
+    );
+    console.log("incrementResponse", incrementResponse);
```
Comment (hugocaillard) [inline examples/counter/tests/counter.test.ts:27]:
```suggestion
```

File: examples/lightning-swaps/contracts/stxswap_v10.clar [hunk 11]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (hugocaillard) [inline examples/lightning-swaps/contracts/stxswap_v10.clar:1]:
```suggestion
```

**Review by hugocaillard — CHANGES_REQUESTED** [review]
just a couple of remaining logs

**Comment by beguene** [comment]
Thanks a lot @hugocaillard !

**Review by hugocaillard — APPROVED** [review]
LGTM

**Pull request merged**